qubes/tests/integ/vm_update.py

```diff
@@ -412,49 +420,200 @@ def test_020_updates_available_notification(self):
+
+        :type self: qubes.tests.SystemTestCase | VmUpdatesMixin
+        :type expected_exit_code: int
+        """
+        logpath = os.path.join(self.tmpdir, 'vm-update-output.txt')
+        with open(logpath, 'w') as f_log:
+            proc = self.loop.run_until_complete(asyncio.create_subprocess_exec(
+                'qubes-vm-update', "--targets", self.testvm1.name,
+                stdout=f_log,
+                stderr=subprocess.STDOUT))
+        self.loop.run_until_complete(proc.wait())
+        if proc.returncode != expected_exit_code:
+            del proc
+            with open(logpath) as f_log:
+                self.fail("qubes-vm-update failed: " + f_log.read())
+        del proc
+
+    def test_110_update_via_proxy_qubes_vm_update(self):
+        """
+        Test both whether updates proxy works and whether is actually used
+        by the qubes-vm-update
+
+        :type self: qubes.tests.SystemTestCase | VmUpdatesMixin
+        """
+        if self.template.count("minimal"):
+            self.skipTest("Template {} not supported by this test".format(
+                self.template))
+
+        self.start_vm_with_proxy_repo()
+
+        with self.qrexec_policy(
+                'qubes.UpdatesProxy', self.testvm1, '$default',
+                action='allow,target=' + self.netvm_repo.name):
+            self.install_test_package()
+            self.add_update_to_repo()
+            self.run_qubes_vm_update_and_assert(expected_exit_code=0)
+
+    def test_120_updates_available_notification_qubes_vm_update(self):
+        """
+        Test if updates-available flags is updated.
+
+        :type self: qubes.tests.SystemTestCase | VmUpdatesMixin
+        """
+        self.start_standalone_vm_with_repo()
+
+        self.upgrade_status_notify()
+        self.assertFalse(self.testvm1.features.get('updates-available', False))
+
+        self.install_test_package()
+        self.assertFalse(self.testvm1.features.get('updates-available', False))
+
         self.add_update_to_repo()
         # update repository metadata
         self.assertRunCommandReturnCode(
             self.testvm1, self.update_cmd, self.exit_code_ok)
 
-        self.loop.run_until_complete(
-            self.testvm1.run_for_stdio(
-                '/usr/lib/qubes/upgrades-status-notify',
-                user='root',
-            ))
+        self.upgrade_status_notify()
         self.assertTrue(self.testvm1.features.get('updates-available', False))
 
-        # install updates
-        self.assertRunCommandReturnCode(
-            self.testvm1, self.upgrade_cmd, self.exit_code_ok)
+        self.run_qubes_vm_update_and_assert(expected_exit_code=0)
 
         self.assertFalse(self.testvm1.features.get('updates-available', False))
 
+    def detach_netvm(self):
+        """
+        Detach net vm to enforce connection error.
+
+        :type self: qubes.tests.SystemTestCase | VmUpdatesMixin
+        """
+        self.loop.run_until_complete(self.testvm1.shutdown())
```

Comment:
Changing netvm dynamically is supported, so you don't necessarily need to shut it down. But if you still want to change netvm while VM is not running, then you probably want to add `wait=True`.